### DISCUSSION TITLE
RSA Verification with cosign

### DISCUSSION THREAD

**Discussion by zosocanuck** [sse-secure-systems/connaisseur#413] [Ideas]
Hi,

cosign v1.3.0 now supports RSA public key verification as part of the `cosign -verify -key` operation, and therefore it would be great for connaisseur to add RSA public verification for cosign, in addition to ECDSA.  I've gone ahead and created a specific version at https://github.com/zosocanuck/connaisseur which may be of interest.  Happy to open a PR if this is a direction you'd like to take.

**Comment by xopham** [comment]
Hi @zosocanuck , good point! Thanks for sharing the reference implementation. Do you actually use RSA in your case? If so, why?
Think we should clean up that section to improve code quality for the different types of keys, as it starts to be a bit messy especially as we are adding #407 . We'll need to look into it.
(linked to #201 )

**Comment by zosocanuck** [comment]
Yes, I work with enterprise customers and there is still consistent demand for the use of the RSA algorithm for signatures and verification.  The majority are Elliptic curve however IMO there should be basic support for RSA given the deployment or RSA within enterprise signing environments.

**Reply by xopham** [reply]
thanks for sharing those insights :pray: based on your work, I created a first tentative integration of rsa in #416 . Does that work for you? Would be nice to get your feedback given the current lack of an e2e test on our side.

**Reply by xopham** [reply]
@zosocanuck do you have a minimal example for creating a valid cosign rsa signature?

**Reply by zosocanuck** [reply]
You can now use cosign (main branch) to import RSA keypairs to produce and verify RSA signatures: https://github.com/sigstore/cosign/pull/1050
